src/lib/migration/proposal_store.rb

```diff
@@ -0,0 +1,230 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2014 Novell, Inc. All Rights Reserved.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+module Migration
+  # 1. Provides access to metadata of proposal parts (clients), as defined in the control file elements
+  # /productDefines/proposals/proposal: https://github.com/yast/yast-installation-control/blob/master/control/control.rnc
+  # 2. Handles all calls to the parts (clients).
+  class ProposalStore < Installation::ProposalStore
+    include Yast::Logger
+    include Yast::I18n
+
+    # @ param[String] proposal_mode one of initial, service, network, hardware,
+    #   uml, ... or anything else
+    def initialize(proposal_mode)
+      textdomain "installation"
+
+      @proposal_mode = proposal_mode
```

Comment:
proposal mode is not needed only to be passed to below class. so simple `super(proposal_mode)` is enough